packages/client/src/links/httpSubscriptionLink.ts

```diff
@@ -79,22 +88,22 @@ export function unstable_httpSubscriptionLink<
             // already unsubscribed - rare race condition
             return;
           }
-          eventSource = new EventSource(url, eventSourceOptions);
+
+          eventSource = new EventSourceWrapper(url, eventSourceOptions);
+
           const onStarted = () => {
             observer.next({
               result: {
                 type: 'started',
               },
               context: {
-                eventSource,
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                eventSource: eventSource!.getEventSource(),
               },
             });
-
-            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-            eventSource!.removeEventListener('open', onStarted);
           };
-          // console.log('starting', new Date());
-          eventSource.addEventListener('open', onStarted);
+          eventSource.addEventListener('open', onStarted, { once: true });
```

Comment:
Important spec thing given the change in how this is run "once". When we do reinitialize, this callback will be called again with the new EventSource instance.

That felt right in-case anybody depends on holding that instance so they can update the reference and re-attach their own listeners